src/tests/performance/connections.rs

```diff
@@ -0,0 +1,217 @@
+            // Synthetic node runs until it completes or is instructed to exit
+            synth_handles.push(tokio::spawn(async move {
+                tokio::select! {
+                    _ = exit_rx => {},
+                    _ = simulate_peer(node_addr, synth_handshaken) => {},
+                };
+            }));
+        }
+
+        // Wait for all peers to indicate that they've completed the handshake portion
+        // or the iteration timeout is exceeded.
+        let _ = tokio::time::timeout(MAX_ITER_TIME, async move {
+            for _ in 0..synth_count {
+                handshake_rx.recv().await.unwrap();
+            }
+        })
+        .await;
+
+        // Send stop signal to peer nodes. We ignore the possible error
+        // result as this will occur with peers that have already exited.
+        for stop in synth_exits {
+            let _ = stop.send(());
+        }
+
+        // Wait for peers to complete
+        for handle in synth_handles {
+            handle.await.unwrap();
+        }
+
+        // Collect stats for this run
+        let mut stats = Stats::new(MAX_PEERS, synth_count);
+        stats.time = test_start.elapsed().as_secs_f64();
+        {
+            let snapshot = test_metrics.take_snapshot();
+
+            stats.accepted = snapshot.get_counter(METRIC_ACCEPTED) as u16;
+            stats.terminated = snapshot.get_counter(METRIC_TERMINATED) as u16;
+            stats.rejected = snapshot.get_counter(METRIC_REJECTED) as u16;
+            stats.conn_error = snapshot.get_counter(METRIC_ERROR) as u16;
+
+            stats.timed_out = synth_count - stats.accepted - stats.rejected - stats.conn_error;
+        }
+        all_stats.push(stats);
+    }
+
+    node.stop().unwrap();
+
+    // Display results table
+    println!("\r\n{}", fmt_table(Table::new(&all_stats)));
+}
+
+async fn simulate_peer(node_addr: SocketAddr, handshake_complete: Sender<()>) {
+    let config = TestConfig::default();
+    let mut synth_node = SyntheticNode::new(&config).await;
+
+    // Establish peer connection
+    let handshake_result = synth_node.connect(node_addr).await;
+    handshake_complete.send(()).await.unwrap();
+    match handshake_result {
+        Ok(stream) => {
+            metrics::counter!(METRIC_ACCEPTED, 1);
+            stream
+        }
+        Err(_err) => {
+            metrics::counter!(METRIC_REJECTED, 1);
+            return;
+        }
+    };
+
+    // Keep connection alive by consuming messages
+    loop {
+        match synth_node
+            .recv_message_timeout(Duration::from_millis(100))
+            .await
+        {
+            Ok((_, _message)) => continue, // consume every message ignoring it
+            Err(_timeout) => {
+                // check for broken connection
+                if !synth_node.is_connected(node_addr) {
+                    metrics::counter!(METRIC_TERMINATED, 1);
+                    synth_node.shut_down().await;

```

Comment:
nice!
this could be done in our [zcash impl](https://github.com/runziggurat/zcash/blob/main/src/tests/performance/connections.rs#L291) later as well probably